### PR TITLE
GFXObjects use smart_ptr [Vulkan][Device]

### PR DESCRIPTION
**Pull request by bluesky013**
Re: #

### Changelog

* GFXObjects use smart_ptr [Vulkan][Device]

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/renderer/gfx-vulkan/VKCommandBuffer.cpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ void CCVKCommandBuffer::doInit(const CommandBufferInfo & /*info*/) {
 }
 
 void CCVKCommandBuffer::doDestroy() {
-    CC_SAFE_DELETE(_gpuCommandBuffer);
+    _gpuCommandBuffer = nullptr;
```
Comment (minggo) [inline native/cocos/renderer/gfx-vulkan/VKCommandBuffer.cpp:67]: IntrusivePtr also has `reset()`.

**Review by minggo — APPROVED** [review]

File: native/cocos/renderer/gfx-vulkan/VKCommandBuffer.cpp (hunk 1 above)
Comment (bluesky013) [reply native/cocos/renderer/gfx-vulkan/VKCommandBuffer.cpp:67]: I took a look at the implementation of `unique_ptr`.  It overloads `operator=(nullptr_t)`. I can modify to the same style.
```cpp
unique_ptr& operator=(nullptr_t) noexcept {
    reset();
    return *this;
}
```

Comment (minggo) [reply native/cocos/renderer/gfx-vulkan/VKCommandBuffer.cpp:67]: What do you want to modify?

Comment (bluesky013) [reply native/cocos/renderer/gfx-vulkan/VKCommandBuffer.cpp:67]: I prefer to replace `reset()` with nullptr assginment. `reset` would be a bit confusing if we don't know it is a smart_ptr or the class has a `reset` member function.

Comment (minggo) [reply native/cocos/renderer/gfx-vulkan/VKCommandBuffer.cpp:67]: Got it.

File: native/cocos/renderer/gfx-vulkan/VKDevice.cpp [hunk 2]
```diff
@@ -504,11 +502,6 @@ void CCVKDevice::doDestroy() {
 
     SPIRVUtils::getInstance()->destroy();
 
-    for (CCVKTexture *texture : _depthStencilTextures) {
-        CC_SAFE_DESTROY_AND_DELETE(texture)
-    }
-    _depthStencilTextures.clear();
-
     if (_gpuDevice) {
```
Comment (star-e) [inline native/cocos/renderer/gfx-vulkan/VKDevice.cpp:505]: We might still call `_depthStencilTextures.clear()` to keep the current destruction order.
In some cases, destruction order is purposefully designed by the author, so changing order might result in bugs.

Comment (bluesky013) [reply native/cocos/renderer/gfx-vulkan/VKDevice.cpp:505]: I deleted the `_depthStencilTextures` in `VKDevice`. It doesn't seem to be used.

Comment (star-e) [reply native/cocos/renderer/gfx-vulkan/VKDevice.cpp:505]: I see.

**Review by star-e — APPROVED** [review]

**Comment by bluesky013** [comment]
@cocos-robot run test cases

**Pull request merged**